api/apps/v1alpha1/application_defaults.go

```diff
@@ -70,25 +71,42 @@ func (in *Scenario) Default() {
 }
 
 func (in *Objective) Default() {
+	for i := range in.Goals {
+		in.Goals[i].Default()
+	}
+
+	if in.Name == "" {
```

Comment:
Your aversion for statement nesting is matched only by my aversion to commit diffs that appear unnecessarily complex because they change indentation :)